src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -519,9 +520,13 @@ export default {
 			}, 500)
 		},
 		debounceFetchSearchResults: debounce(function() {
+			this.resetNavigation()
 			if (this.isSearching) {
 				this.fetchSearchResults()
 			}
+			this.$nextTick(() => {
+				this.initializeNavigation('.list-item')
+			})
 		}, 250),
```

Comment:
so whenever a character is entered as input, all event listeners are removed and then added again to almost the same items ? is it better maybe to compare the previous `itemElements` and skip the common ones ?

Reply:
Order of items could change, which will break a flow (item with index 2 could become 6th, and focus will jump weirdly)
So it's safe to remove old listeners and add new ones, it shouldn't load browser too much for 10-20 items

Reply:
The indexes will follow the new `itemElements` hence the order is correct. I am thinking about cases where a user can have +50 items for example.

Reply:
> Order of items could change, which will break a flow (item with index 2 could become 6th, and focus will jump weirdly) So it's safe to remove old listeners and add new ones, it shouldn't load browser too much for 10-20 items

Can we listen the keydown event not on items itself but on a whole container?

Reply:
Or create vue component with `v-on` directive instead of manually query elements and manage listeners. Then Vue will handle optimizations around managing listeners itself.

*Universal component, with a scoped-slot

Reply:
I think that it's better to keep navigation logic in a separate composable, and not duplicate it for each component we need. Currently we use it only for two item lists, but it could be every list in the application.
In addition, that container should also be able to perform other functions (be a virtual scroller, for example). In that case, we need also to reinvent a way to work for each case.